### PR TITLE
Add a printer job queue via the printer plugin

### PR DESCRIPTION
**Pull request by paroga**

### CODE REVIEW

File: plugins/printer/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* `aborted`: the printer stopped the action
+* `completed`: print was successful
+
+**Example**:
+A server sending `[12,16]` via WebSocket indicates that the two documents `12`
+and `16` are ready for printing. The client will request the first document
+from `/foodcoop/printer/12` and send it to the printer. The status can be
+updated by sending `{"id":12,"state":"pending"}` via WebSocket to the server.
+Sending `{"id":12,"state":"completed","finish":true}` as soon as when the
+printing succeded will mark the job done.
```
[the remark this answers is not in the document]
Comment (paroga) [reply plugins/printer/README.md]: i'd just change it from `[1,2,3]` to `{"unfinished_jobs":[1,2,3]}`, which should allow enough extendability for the future

File: plugins/printer/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+This plugin is enabled by default in foodsoft, so you don't need to do anything
+to install it. If you still want to, for example when it has been disabled,
+add the following to foodsoft's Gemfile:
+
+```Gemfile
+gem 'foodsoft_printer', path: 'plugins/foodsoft_printer'
+```
+
+This plugin introduces the foodcoop config option `printer_token`, which takes
+the a random string for authentication at the endpoint. Additionally a set of
```
[the remark this answers is not in the document]
Comment (paroga) [reply plugins/printer/README.md]: i don't see it ^^, can you point me to the correct word

**Comment by paroga** [comment]
> And perhaps we need to think about when we enable a plugin by default, and when not (only when we have an infrastructure to enable plugins easily from a Docker build step).

I'd just enable all plugins via `Gemfile`, since they all have their own `use_XXX` option anyway.

> Note to self: request testing for webserver change when releasing a new version of Foodsoft.

Deploying with `puma` resulted in a lot of strange MySQL (concurrency?) errors, when deployed in `production`. `development` (with SQLite) worked without problems. I had to deploy a second instance only for the `/printer/socket` endpoint to get it working. Personally I don't want to put too much effort into tracking down the problems since I prefer to port it to `ActionCable` anyway when we going to update our `rails` version...

File: plugins/printer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (wvengen) [reply plugins/printer/README.md]: `which takes the a random` -> `which takes a random`

**Comment by paroga** [comment]
I've added a comment about puma, but will add it to the `Gemfile` in a separate PR.

**Comment by wvengen** [comment]
I do feel a little uneasy about the server requirements. I don't think we should enable this by default, if it requires special setup because of the webserver ...

**Comment by paroga** [comment]
agree. i've only added a comment in `Gemfile`

**Pull request merged**